packages/labs/task/src/task.ts

```diff
@@ -102,71 +113,92 @@ export class Task<T extends [...unknown[]] = any, R = any> {
    * is kept and only resolved when the new run is completed.
    */
   taskComplete!: Promise<R>;
+
+  /**
+   * Controls if they task will run when its arguments change. Defaults to true.
+   */
+  autoRun = true;
+
   private _resolveTaskComplete!: (value: R) => void;
   private _rejectTaskComplete!: (e: unknown) => void;
 
   constructor(
     host: ReactiveControllerHost,
-    task: TaskFunction<T, R>,
-    getDependencies: DepsFunction<T>
+    task: TaskFunction<T, R> | TaskConfig<T, R>,
+    args?: ArgsFunction<T>
   ) {
     this._host = host;
     this._host.addController(this);
+    if (typeof task === 'object') {
+      const taskConfig = task as TaskConfig<T, R>;
+      args = taskConfig.args;
+      task = taskConfig.task;
+      if (taskConfig.autoRun !== undefined) {
+        this.autoRun = taskConfig.autoRun;
+      }
+    }
     this._task = task;
-    this._getDependencies = getDependencies;
+    this._getArgs = args;
     this.taskComplete = new Promise((res, rej) => {
```

Comment:
Should this change for manual run tasks so that it's a resolved promise?

Reply:
Can you explain why?

Reply:
Just wondering what `await task.taskComplete` should do for a manually run task? The task is idle, but it's not necessarily ever going to run. In some ways it seems like a task that has completed some times, and might or might not run again, and in that case `taskComplete` would be a resolved promise.

Reply:
Let's handle this as a follow on. Filed https://github.com/lit/lit/issues/2347.